internal/storage/payload_writer_test.go

```diff
@@ -292,11 +290,39 @@ func TestPayloadWriter_Failed(t *testing.T) {
 		err = w.AddFloatToPayload(data, nil)
 		require.Error(t, err)
 
-		w, err = NewPayloadWriter(schemapb.DataType_Int64, false)
+		w, err = NewPayloadWriter(schemapb.DataType_Int64)
 		require.Nil(t, err)
 		require.NotNil(t, w)
 
 		err = w.AddFloatToPayload(data, nil)
 		require.Error(t, err)
 	})
 }
+
+func TestParquetEncoding(t *testing.T) {
+	t.Run("test int64 pk", func(t *testing.T) {
+		field := &schemapb.FieldSchema{IsPrimaryKey: true, DataType: schemapb.DataType_Int64}
+
+		w, err := NewPayloadWriter(schemapb.DataType_Int64, WithWriterProps(getFieldWriterProps(field)))
+
+		assert.NoError(t, err)
+		err = w.AddDataToPayload([]int64{1, 2, 3}, nil)
+		assert.NoError(t, err)
+
+		err = w.FinishPayloadWriter()
```

Comment:
If a certain encoding type is expected, there should be an assertion. A simple assertion on the artifact size would do.